data/transactions/logic/eval.go

```diff
@@ -1568,9 +1621,7 @@ func opTxn(cx *evalContext) {
 		cx.err = fmt.Errorf("invalid txn field %d", field)
 		return
 	}
-	var sv stackValue
-	var err error
-	sv, err = cx.txnFieldToStack(&cx.Txn.Txn, field, 0, cx.GroupIndex)
+	sv, err := cx.txnFieldToStack(&cx.Txn.Txn, field, 0, cx.GroupIndex)
```

Comment:
I think this should be in `gtxna` since `txn` returns self fields.

Plus, Scratch is not a real txn field, so we might want a new `gscratch` or `gload` opcode. @jannotti thoughts?

Reply:
I think it's cleanup that exists across many of the txn, txna, gtxn, etc. functions, but you are correct that I only modified this one. I'll fix that.

Reply:
to access a scratch item there are few bits of info needed: txn index in a group and scratch item position. The only one opcode allows this: `gtxna` - it is capable to read an item from an array by position from a different txn by index.

Reply:
Yes - this change is purely cleanup and doesn't affect the implementation at all.

Reply:
I am somewhat in favor of `gload <txn> <scratch>` though iand we'd also need `gloads <scratch>` which would take the txn slot from stack. (but still take the scratch slot from immediate arg)

All in all, I think it would probably be better.  It would be easier to document (we don't have a convenient way to indicate Scratch is a stateful only field), but we have a standard way to say an opcode is stateful only.